src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2062 @@
+        text_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindTextModel`].
+        vision_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindVisionModel`].
+        audio_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`ImageBindAudioModel`].
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    logits_per_image: torch.FloatTensor = None
+    logits_per_text: torch.FloatTensor = None
+    logits_per_audio: torch.FloatTensor = None
+    image_embeds: torch.FloatTensor = None
+    text_embeds: torch.FloatTensor = None
+    audio_embeds: torch.FloatTensor = None
+    vision_model_output: BaseModelOutputWithPooling = None
+    text_model_output: BaseModelOutputWithPooling = None
+    audio_model_output: BaseModelOutputWithPooling = None
+
+    def to_tuple(self) -> Tuple[Any]:
+        fields_to_exclude = [
+            "text_model_output",
+            "vision_model_output",
+            "audio_model_output",
+        ]
+        return tuple(self[k] if k not in fields_to_exclude else getattr(self, k).to_tuple() for k in self.keys())
+
+
+class ImageBindGenericPatchEmbedding(nn.Module):
+    """Generic Patch Embedding class that can be used for Vision (image/video), Audio, Depth, Thermal modalities."""
+
+    def __init__(
+        self,
+        config: Union[ImageBindVisionConfig, ImageBindAudioConfig],
+        projection: nn.Module,
+        use_layernorm: bool = False,
+    ):
+        super().__init__()
+
+        if hasattr(config, "image_size"):
+            image_size = config.image_size
+        elif hasattr(config, "num_mel_bins") and hasattr(config, "target_len"):
+            image_size = (config.num_mel_bins, config.target_len)
+        else:
+            raise ValueError("Either `image_size` or `num_mel_bins` and `target_len` must be provided in the config.")
+
+        self.image_size = image_size if isinstance(image_size, collections.abc.Iterable) else (image_size, image_size)
+        self.num_channels = config.num_channels
+
+        self.projection = projection
+        self.layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps) if use_layernorm else None
+
+    def forward(self, input_values: torch.FloatTensor, interpolate_pos_encoding: bool = False) -> torch.Tensor:
+        if input_values.ndim not in [4, 5]:
+            raise ValueError(f"Input tensor shape should have length 4 or 5 but got {input_values.ndim}.")
+
+        _, num_channels, *spatial_shape = input_values.shape
+        height, width = spatial_shape[-2:]
+
+        if num_channels != self.num_channels:
+            raise ValueError(
+                "Make sure that the channel dimension of the pixel values match with the one set in the configuration."
+                f" Expected {self.num_channels} but got {num_channels}."
+            )
+        if not interpolate_pos_encoding:
```

Comment:
no seeing the interpolate used here, (if it is used in another layer do you have to check here?)

Reply:
`interpolate_pos_encoding` is used [here](https://github.com/huggingface/transformers/blob/58e1c3a951b800694c1ecfd49f9edc5c9670deed/src/transformers/models/imagebind/modeling_imagebind.py#L315) for `VisionEmbeddings`. But the var `interpolate_pos_encoding` is used to check in interpolation has been performed or not and if not then the height and width should remain the same